docs/capabilities.md

```diff
@@ -0,0 +1,78 @@
+# Capabilities of an user required for a ceph-csi in an Ceph cluster
+
+Ceph uses the term capabilities to describe authorizing an authenticated user
+to exercise the functionality of the monitors, OSDs and metadata servers.
+Capabilities can also restrict access to data within a pool or pool namespace.
+A Ceph administrative user sets a user's capabilities when creating or
+updating a user.
+
+## RBD
+
+We have provisioner, controller expand and node stage secrets in storage class.
+For the provisioner and controller expand stage secret in storageclass, the
+user needs to have the  below mentioned ceph capabilities.
+
+```text
+"mon", "profile rbd",
+"mgr", "allow rw",
+"osd", "profile rbd"
+```
+
+And for the node stage secret in storageclass, the user needs to have the
+below mentioned ceph capabilities.
+
+```text
+"mon", "profile rbd",
+"osd", "profile rbd",
+"mgr", "allow rw"
+```
+
+## CephFS
+
+Similarly in CephFS, for the provisioner and controller expand stage secret in
+storageclass, the user needs to have the below mentioned ceph capabilities.
+
+```text
+"mon", "allow r",
+"mgr", "allow rw",
+"osd", "allow rw tag cephfs metadata=*"
+```
+
+And for node stage secret in storageclass, the user needs to have
+the below mentioned ceph capabilities.
+
+```text
+"mon", "allow r",
+"mgr", "allow rw",
+"osd", "allow rw tag cephfs *=*",
+"mds", "allow rw"
+```
+
+To get more insights on capabilities of cephfs you can refer
+[this document](https://ceph.readthedocs.io/en/latest/cephfs/client-auth/)
+
+## Command to a create user with required capabilities
+
+### command for RBD
+
+```bash
+ceph auth get-or-create client.kubernetes mon 'profile rbd' osd 'profile rbd' \
```

Comment:
to make it easier to match the above listed capabilities, please re-format the command like this:
```bash
ceph auth get-or-create client.kubernetes \
    mon 'profile rbd' \
    osd 'profile rbd' \
    mgr 'allow rw'
```